build/cmake/GeneratePkgConfig.cmake

```diff
@@ -0,0 +1,298 @@
+    if(NOT DEFINED inst_DESTINATION)
+        # Install based on the libdir:
+        set(inst_DESTINATION "${gx_libdir}/pkgconfig")
+    endif()
+    if(NOT DEFINED inst_RENAME)
+        set(inst_RENAME "${ARG_FILENAME}")
+    endif()
+    # install(CODE) will write a simple temporary file:
+    set(inst_tmp "${CMAKE_CURRENT_BINARY_DIR}/${target}-pkg-config-tmp.txt")
+    _genex_escape(esc_cond "${ARG_CONDITION}")
+    string(CONFIGURE [==[
+        $<@gx_cond@:
+            # Installation of pkg-config for target @target@
+            message(STATUS "Generating pkg-config file: @inst_RENAME@")
+            file(READ [[@gx_tmpfile@]] content)
+            # Insert the install prefix:
+            string(REPLACE "%INSTALL_PLACEHOLDER%" "${CMAKE_INSTALL_PREFIX}" content "${content}")
+            # Write it before installing again:
+            file(WRITE [[@inst_tmp@]] "${content}")
+        >
+        $<$<NOT:@gx_cond@>:
+            # Installation was disabled for this generation.
+            message(STATUS "Skipping install of file [@inst_RENAME@]: Disabled by CONDITION “@esc_cond@”")
+        >
+    ]==] code @ONLY)
+    install(CODE "${code}")
+    _bind_genex_to_target(gx_dest ${target} "${inst_DESTINATION}")
+    _bind_genex_to_target(gx_rename ${target} "${inst_RENAME}")
+    # Wrap the filename to install with the same condition used to generate it. If the condition
+    # is not met, then the FILES list will be empty, and nothing will be installed.
+    install(FILES "$<${gx_cond}:${inst_tmp}>"
+            DESTINATION ${gx_dest}
+            RENAME ${gx_rename}
+            ${inst_UNPARSED_ARGUMENTS})
+endfunction()
+
+# Generates the actual content of a .pc file.
+function(_generate_pkg_config_content out)
+    cmake_parse_arguments(PARSE_ARGV 1 ARG "" "PREFIX;NAME;VERSION;DESCRIPTION;GENEX_TARGET;LIBDIR" "")
+    if(ARG_UNPARSED_ARGUMENTS)
+        message(FATAL_ERROR "Unknown arguments: ${ARG_UNPARSED_ARGUMENTS}")
+    endif()
+    set(content)
+    string(APPEND content
+        "# pkg-config .pc file generated by CMake ${CMAKE_VERSION} for ${ARG_NAME}-${ARG_VERSION}. DO NOT EDIT!\n"
+        "prefix=${ARG_PREFIX}\n"
+        "exec_prefix=\${prefix}\n"
+        "libdir=\${exec_prefix}/${gx_libdir}\n"
+        "\n"
+        "Name: ${ARG_NAME}\n"
+        "Description: ${ARG_DESCRIPTION}\n"
+        "Version: ${ARG_VERSION}"
+        )
+    # Add Requires:
+    set(requires_joiner "\nRequires: ")
+    set(gx_requires $<GENEX_EVAL:$<TARGET_PROPERTY:pkg_config_REQUIRES>>)
+    set(has_requires $<NOT:$<STREQUAL:,${gx_requires}>>)
+    string(APPEND content "$<${has_requires}:${requires_joiner}$<JOIN:${gx_requires},${requires_joiner}>>\n")
+    string(APPEND content "\n")
+    # Add "Libs:"
+    set(libs)
+    # Link options:
+    set(gx_libs
+        "-L\${libdir}"
+        "-l$<TARGET_PROPERTY:OUTPUT_NAME>"
+        $<GENEX_EVAL:$<TARGET_PROPERTY:pkg_config_LIBS>>
+        $<TARGET_PROPERTY:INTERFACE_LINK_OPTIONS>
```

Comment:
> If you build with instrumentation (coverage, sanitizers), the installed pkg-config files will now include the necessary link flags as usage requirements.

I do not think this currently works as intended. Building with `-DMONGO_SANITIZE=address,undefined` does not inject `-fsanitize=address,undefined` into the the LFLAGS list as it should. Other dependent libraries (i.e. resolv) are also not present. Consider using `$<TARGET_PROPERTY:INTERFACE_LINK_LIBRARIES>` with appropriate parsing instead?

Reply:
Oh oop. This is both right and wrong. The sanitize flags are not propagated because they are not added as interface properties on the targets that need them, but I have an upcoming set of changes that will make that work correctly, and I forgot that I hadn't yet PR'd the other changes.